WcaOnRails/app/models/competition.rb

```diff
@@ -850,6 +851,14 @@ def pending_results_or_report(days)
     if refund_policy_limit_date? && refund_policy_limit_date > start_date
       errors.add(:refund_policy_limit_date, I18n.t('competitions.errors.refund_date_after_start'))
     end
+
+    if modify_registered_events < registration_close
+      errors.add(:modify_registered_events, I18n.t('competitions.errors.event_modification_before_reg_close'))
+      end
+    
+    if on_the_spot_registration == true && modify_registered_events =! start_date
```

Comment:
`if on_the_spot_registration == true` is simply `if on_the_spot_registration`.

Be careful: `a =! b` doesn't actually do what you want!
It will affect the value `!b` to variable `a`.
If you want to check if `a` is different from `b` you want to use `a != b`.